Regions

uniphil opened the pull request.

uniphil committed:
Show regions from static topojson

uniphil committed:
Load schools for regions

so we can compute stuff

uniphil committed:
Start moving colouring stuff into place

uniphil committed:
Colourize regions

uniphil committed:
Implement regions hover

uniphil committed:
Hide some sidebar stuff in region view

uniphil committed:
Recolour regions for new schools data

uniphil committed:
Style region on hover

esoliani commented on app/scripts/controllers/dashboard.coffee:
```diff
@@ -10,11 +10,13 @@
+    'staticApi', 'watchComputeSrv',
 
     ($scope, $window, $routeParams, $anchorScroll, $http, leafletData,
     _, $q, WorldBankApi, layersSrv, chartSrv, $log, $location, $translate,
```
chart server is removed in https://github.com/WorldBank-Transport/edudash/pull/116. we might have some problems

uniphil replied:
I think since `chartSrv` was on a different line (L12), they won't conflict.

esoliani commented on app/scripts/controllers/dashboard.coffee:
```diff
@@ -75,10 +80,44 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
+                  for region, schools of schoolsByRegion
+                    detailsByRegion[region] =
+                      # TODO: should these averages be weighted by number of pupils?
+                      AVG_GPA: averageProp schools, 'AVG_GPA'
```
the first two fields are from secondary and the last are from primary.

uniphil replied:
Yeah, they become `NaN` if the properties are missing, and later code checks schoolType so the right ones are used. Probably better to check and only assign the valid ones here though, I'll add it.

esoliani commented on app/scripts/controllers/dashboard.coffee:
```diff
@@ -75,10 +80,44 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
+                      AVG_GPA: averageProp schools, 'AVG_GPA'
+                      CHANGE_PREVIOUS_YEAR_GPA: averageProp schools, 'CHANGE_PREVIOUS_YEAR_GPA'
+                      AVG_MARK: averageProp schools, 'AVG_MARK'
+                      CHANGE_PREVIOUS_YEAR_MARK: 'CHANGE_PREVIOUS_YEAR_MARK'  # TODO: confirm
```
typo: the field name is 'CHANGE_PREVIOUS_YEAR', we need to fix it on many places

uniphil committed:
Compute region propertes according to schoolType

esoliani commented on app/views/dashboard.html:
```diff
@@ -98,8 +98,8 @@ <h4 translate="legend.ptratio">Pupil-Teacher Ratio</h4>
-          <li role="presentation" ng-class="{active: viewMode==='regional'}">
-            <a ng-click="setViewMode('regional')">
+          <li role="presentation" ng-class="{active: viewMode==='regions'}">
+            <a ng-click="setViewMode('regions')">
```
how to unset the region viewMode?

uniphil replied:
Right now you can click on "Primary Schools" again to get back to the school view (if you're looking at primary schools). We can make this button a toggle as well if you want.

esoliani commented on app/views/regionFlyout.html:
```diff
@@ -1,18 +1,18 @@
 <header>
   <h2>Region Name</h2>
```
we need to add translate for this string

uniphil replied:
good catch

uniphil committed:
Add missing translations

uniphil committed:
Fix wiredep for bootstrap css

Bootstrap changed their `bower.json`'s `main` array, so fix it for us.
We don't use any bootstrap javascript, so leave it out.

uniphil committed:
Use empty object if not window.topojson

Might save some tests from breaking

uniphil committed:
Merge branch 'origin/edudash-2.0' into polygons

uniphil committed:
Put bootstrap's js back in

It's used for the year selector
